doc/operations/code_upgrade_1x.rst

```diff
@@ -0,0 +1,44 @@
+Code Upgrade 1.x from 2.0 Guidelines
+====================================
+
+.. note:: The generic code upgrade procedure is described :doc:`here </operations/code_upgrade>`.
+
+The CCF 2.0 release introduces major improvements and new features from 1.x LTS releases. To be able to upgrade the service as smoothly as possible and make the most of these new features, operators should follow this specific guide to upgrade their 1.x service.
+
+New ``cchost`` JSON Configuration
+---------------------------------
+
+The configuration for a single CCF node (specific to ``cchost`` binary) is now a single JSON file, rather than a set of command line arguments or INI/YAML configuration file (see :doc:`/operations/configuration` for more details).
+
+The ``migrate_1_x_config.py`` Python script, available as part of the :doc:`ccf Python package </audit/python_library>`, converts an existing 1.x ``.ini`` configuration file to a 2.0 JSON configuration file:
+
+.. code-block:: bash
+
+    $ pip install ccf
+    $ migrate_1_x_config.py config_1_x.ini 2_x_config.json # migrate_1_x_config.py is in path
+    2022-01-01 10:00:00.000 | INFO     | __main__:<module>:69 - Reading 1.x configuration file: config_1_x.ini
+    2022-01-01 10:00:00.000 | DEBUG    | __main__:<module>:77 - Found sections: ['default', 'start', 'join', 'recover']
+    2022-01-01 10:00:00.000 | INFO     | __main__:<module>:98 - Command type: start
+    2022-01-01 10:00:00.000 | SUCCESS  | __main__:<module>:215 - JSON configuration successfully written to: 2_x_config.json
+
+The ``cchost`` executable can be used to verify the validity of the migrated configuration file (without launching the enclave application):
+
+.. code-block:: bash
+
+    $ /opt/ccf/bin/cchost --config 2_x_config.json --check
+    2022-01-01T10:00:00.000000Z        100 [info ] ../src/host/main.cpp:78              | CCF version: ccf-2.0.0
+    2022-01-01T10:00:00.000000Z        100 [info ] ../src/host/main.cpp:95              | Configuration file successfully verified
+
+Upgrade to Latest 1.x LTS before 2.0
+------------------------------------
+
+During the code upgrade process, the service will temporarily be made of nodes from different CCF versions. This means that breaking changes introduced in 2.0 (e.g. ledger format for application claims) introduce incompatibility with the ledger format that 1.x nodes can apply.
+
+To prevent this, operators should first complete the :ref:`code upgrade procedure <operations/code_upgrade:Procedure>` from the current 1.x version the service is running on (e.g. ``1.0.10``) to a version of the 1.x LTS as least as high as `1.0.16 <https://github.com/microsoft/CCF/releases/tag/ccf-1.0.16>`_. Once this is done, the service can be safely upgraded to 2.0.
```

Comment:
To the latest LTS, that's the release compatibility policy, we should avoid hardcoding a version here.